src/main/kotlin/org/elaastic/questions/assignment/sequence/LearnerSequenceService.kt

```diff
@@ -72,4 +77,22 @@ class LearnerSequenceService(
             } else it
         }
 
+    /**
+     * Count the number of reports made by the user for the given sequence
+     *
+     * If an error occurred when we retrieve all the peerGrading, then the
+     * count is 0
+     */
+    fun countReportMade(user: User, sequence: Sequence): Int {
+        val peerGrading = emptyList<PeerGrading>().toMutableList()
+        return try {
+            peerGrading.addAll(peerGradingService.findAllEvaluationMadeForLearner(user, sequence))
+            peerGrading.addAll(peerGradingService.findAllEvaluationMadeForLearner(user, sequence, 2))
+            peerGrading.count { !it.reportReasons.isNullOrBlank() }
+        } catch (_: IllegalStateException) {
```

Comment:
Why 3 requests for that? The last probably induces N+1 select issue.
Only one request is necessary to get this information.

Reply:
Fix in e6f5a8f8e22f38bd99d1d32b8228ce01340d17e3